File: URI/Lua/oescolhido.lua
n = io.read("*number")
maior = io.read("*number")
m_maior = io.read("*number")

for i=2,n
do
    matricula = io.read("*number")
    nota = io.read("*number")
    if nota > m_maior then
        m_maior = nota
        maior = matricula
    end
end

if m_maior < 8 then
    print("Minimum note not reached")
else
    print(maior)
end

File: URI/C/oescolhido.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>
typedef unsigned int uint;
typedef unsigned long long int ulli;
#define true 1
#define false 0

int main(void){
    uint matricula, maior, n;
    float nota, m_nota;
    scanf("%d", &n);
    scanf("%d %f", &matricula, &nota);
    maior = matricula;
    m_nota = nota;

    while (n --> 1)
    {
        scanf("%d %f", &matricula, &nota);
        if (nota > m_nota)
        {
            m_nota = nota;
            maior = matricula;
        }
    }
    if (m_nota < 8)
    {
        puts("Minimum note not reached");
    }
    else
    {
        printf("%d\n", maior);
    }
    return 0;
}